bootloader/assets/assets.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package assets
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+var registeredAssets = map[string][]byte{}
+
+// registerAsset registers a boot asset under the given name.
+func registerAsset(name string, data []byte) {
+	if _, ok := registeredAssets[name]; ok {
+		panic(fmt.Sprintf("asset %v is already registered", name))
+	}
+	registeredAssets[name] = data
+}
+
+// GetBootAsset returns the content of boot asset registered under the given
+// name, or nil when none was found.
+func GetBootAsset(name string) []byte {
```

Comment:
it's under bootloader so we sort of know it's about boot, maybe  s/BootAssert/InternalAsset/

Reply:
InternalBootAsset?

Reply:
To bike shed a little, since the package name is `assets` I would go with `assets.InternalBootAsset` or `assets.BootAsset` if the word internal is skipped.

Reply:
actually, this probably should just be assets.Internal("name")

Reply:
notice that it will be used mostly with constants so it  should be relatively clear what is going on:

assets.Internal("grub.cfg")